### ISSUE TITLE
Teams suffix/prefix limits is based on server version

### ISSUE THREAD

**Issue by f4n74z14** [retrooper/packetevents#960]
**Describe the bug**
If you have packet-based plugins for teams, PacketEvents do not support them (extended limits and etc.) because determines version from the server rather than the client.

**Software brand**
PaperSpigot 1.8.8

**Plugins**
ViaVersion, GrimAC (2.3.67), TAB

**How To Reproduce**
1. Set prefix more than 16 chars in TAB on 1.8.8 server
2. Try decode PlayServerTeam
3. The error will appear

**Expected behavior**
PacketEvents determines version from the client rather than the server.

**Comment by AoElite** [comment]
@f4n74z14 You need to send the error

**Comment by f4n74z14** [comment]
> @f4n74z14 You need to send the error

https://pastebin.com/ZZVpkiKn

**Comment by SamB440** [comment]
How does via handle this?

**Comment by f4n74z14** [comment]
I haven't looked into Via's code, but logically I don't see why the server version is used when writing/reading a client-only packet.

**Comment by SamB440** [comment]
Server version is used in all packets because via expects packets sent from the server to be of that version, likewise packetevents does, otherwise you get errors due to differing client and server versions.

**Comment by f4n74z14** [comment]
You are right! Perhaps this could be done by combining the client and server versions, for example: the packet structure would be server based and the limits would be client based. (I looked how it is done in BungeeCord, there the limits are the same for all - Short.MAX_VALUE).

**Comment by f4n74z14** [comment]
It's also an issue for team name limit.

**Comment by sprytex** [comment]
It was 'Scoreboard: true' in the plugin tab when I joined my 1.8.8 server using version 1.21.4.

**Comment by sprytex** [comment]
tested with via, tab, grim only. @SamB440

**Comment by sprytex** [comment]
The trend has started now you will always have 1 star, every day a few reviews until the developers will not raise their European ass who donate to the ukrainian shits and fuck up their country without Russia

**Comment by sprytex** [comment]
fix or 1 star for forever

**Comment by SamB440** [comment]
I recommend that this issue is closed and not fixed as it violates expected vanilla packets.

**Comment by linsaftw** [comment]
Please fix!! We need to use GrimAC + TAB + Via!

**Comment by SamB440** [comment]
TAB has a setting to mitigate this: `compensate-for-packetevents-bug`

**Comment by linsaftw** [comment]
> TAB has a setting to mitigate this: `compensate-for-packetevents-bug`

I am sadly using an older version because of the removal of the infinite nametags...

**Comment by linsaftw** [comment]
Any news on this ? Running TAB + 1.21 + PacketEvents is currently impossible without spam...
